docs/content/docs/deployment/network_buffer.md

```diff
@@ -0,0 +1,125 @@
+Each of that pools is limited to at most 
+```
+#channels * taskmanager.network.memory.buffers-per-channel + taskmanager.network.memory.floating-buffers-per-gate
+```
+
+The size of the buffer can be configured by setting `taskmanager.memory.segment-size`.
+
+### Input network buffers
+Buffers in the input channel are divided into exclusive and floating buffers.
+Flink attempts to acquire the configured amount of the exclusive buffers in the initialization phase for each channel.
+Exclusive buffers can be used only by one particular channel.
+A channel can request additional floating buffers from a buffer pool shared across all channels belonging to the given input gate.
+Flink treats the configured amount of exclusive and floating buffers as only a recommended values.
+If there are not enough buffers available on the input side, Flink will be able to make a progress with zero exclusive buffers and a single floating buffer.
+
+### Output network buffers
+Unlike the input buffer pool, the output buffers pool have only one type of buffers which it shares all among all subpartitions.
+In order to avoid the excessive data skew, the number of buffers for each subpartition is limited by `taskmanager.network.memory.max-buffers-per-channel`.
+
+Similarly, as on the input side, the configured amount of the exclusive buffers and floating buffers is treated only
+as the recommended values. If there are not enough buffers available, Flink will be able to make a progress with
+only a single exclusive buffer per output subpartition and zero floating buffers.
+
+## Performance 
+### Selecting the buffer size
+As was described before, the buffer collects records in order to optimize network overhead for sending 
+the data portion to the next subtask. The too-small buffer size especially less than the one record 
+doesn't make sense because the overhead is still pretty big and anyway the next subtask should receive 
+all parts of record before the consuming it(as result it leads to low throughput). At the same time, 
+the big buffer size leads to high memory usage, huge checkpoint data(for unaligned checkpoint), 
+the big checkpoint time(for aligned checkpoint). Also, the big buffer size 
+doesn't make sense with small `execution.buffer-timeout` because in this case all(or almost all) 
+sending buffer would be only partially full which means that the allocated memory won't be utilized properly.
+
+### Selecting the buffer count
+The number of buffers is mostly configured by `taskmanager.network.memory.buffers-per-channel`, `taskmanager.network.memory.floating-buffers-per-gate`.
+The first one is important for the input channel and allows to have the exclusive buffers 
+for the channel that influence positively on throughput but negatively on memory usage. At the same time,
+`floating-buffers-per-gate` uses the memory in a more optimal way but it can lead to throughput degradation in case of data skew.
```

Comment:
```suggestion
For best throughput in most cases we recommend to stick with the default values for the number of exclusive and floating buffers. If the amount of in-flight data is causing problems enabling [buffer debloating](#Buffer debloat) is again the easiest and the recommended way of trying to deal with those problems. However if for some reason this doesn't work for you, here are things to consider when tuning the amount of network buffers.

You should adjust the amount of buffers according to your expected throughput (in `bytes/second`). Assigning credits and sending buffers takes some time (around two round trip messages between two nodes). This latency depends on your network, but for the sake of simplicity in the healthy local networks we can assume it's `1ms`. Next we take the buffer size (https://ci.apache.org/projects/flink/flink-docs-master/docs/deployment/config/#taskmanager-memory-segment-size). With those three numbers (buffer round trip time, buffer size and expected throughput) we can calculate the number of required buffers to sustain this throughput by using this simple formula:
number_of_buffers = expected_throughput * buffer_roundtrip / buffer_size
For example, with expected throughput of `320MB/s`, round trip latency of `1ms` and the default memory segment size:
number_of_buffers = 320MB/s * 1ms / 32KB = 10
This is the number of actively used buffers that we need to achieve the expected throughput with the given round trip latency. 

The purpose of floating buffers is to handle the data skew scenario. Ideally number of floating buffers (default 8) plus the exclusive buffers (default 2) that belong to that channel should be able to saturate the network throughput. However sometimes this is either not feasible or even not necessary. Keep in mind that it happens very rarely that only a single channel among all subtasks on your task manager is being used.

The exclusive buffers on the other hand have a purpose to provide a fluent throughput. While one buffer is in transit, the other is being filled up. However with larger setups, the number of exclusive buffers is the dominant factor defining the amount of in-flight data Flink is using. So in case of back pressure in low throughput setups, it might be a good idea to reduce the number of [exclusive buffers](https://ci.apache.org/projects/flink/flink-docs-master/docs/deployment/config/#taskmanager-network-memory-buffers-per-channel).

### Final words

To sum up. Tuning memory configuration for the network in Flink should boil down to just enabling the buffer debloating mechanism. If for some reasons this doesn't work, you can try to tune the buffer debloating, or disable it and manually configure the memory segment size and the number of buffers. For this second scenario we recommend two settings:
1. Default values for the max throughput.
2. Reducing the memory segment size and/or number of exclusive buffers to speed up checkpointing/reduce memory consumption of the network stack.
```

Reply:
Plus adjust formatting/fix links?